benchmarks/metrics.py

```diff
@@ -0,0 +1,49 @@
+import typing as tp
+from abc import ABC
+
+
+class EvalMetric(ABC):
+    __eval_name: str = ""
+
+    def __init__(self, name: str, value: float):
+        self.name = name
+        self.value = value
+
+    @property
+    def eval_name(self) -> str:
+        return self.__eval_name
+
+
+class KNNEvalMetric(EvalMetric):
+    __eval_name = "knn_eval"
+
+
+class LinearEvalMetric(EvalMetric):
+    __eval_name = "linear_eval"
+
+
+class FinetuneEvalMetric(EvalMetric):
+    __eval_name = "finetune_eval"
+
+
+EvalMetrics = tp.List[EvalMetric]
+
+
+def eval_metrics_to_markdown(metrics: EvalMetrics) -> str:
+    lines = []
+    header = f"| Eval Name | Metric Name | Value |"
+    lines.append(header)
+    for metric in metrics:
+        line = f"| {metric.eval_name} | {metric.name} | {metric.value} |"
+        lines.append(line)
+    return "\n".join(lines)
```

Comment:
Let's move this function to the `main.py` files for now even if we have to duplicate it. The goal for the benchmarks is that the code is as simple as possible to make them easy to understand and modify.

We might have to rework some of the benchmarking soon, there were some changes regarding metrics in PyTorch Lightning that could simplify our code as well. AFAIK the `trainer.validate()` functions now return the metrics directly which means we won't have to rely on the callback anymore.